spec.html

```diff
@@ -7413,9 +7413,8 @@ <h1>FunctionAllocate ( _functionPrototype_, _strict_, _functionKind_ )</h1>
       <h1>FunctionInitialize ( _F_, _kind_, _ParameterList_, _Body_, _Scope_ )</h1>
       <p>The abstract operation FunctionInitialize requires the arguments: a function object _F_, _kind_ which is one of (Normal, Method, Arrow), a parameter list Parse Node specified by _ParameterList_, a body Parse Node specified by _Body_, a Lexical Environment specified by _Scope_. FunctionInitialize performs the following steps:</p>
       <emu-alg>
-        1. Assert: _F_ is an extensible object that does not have a `length` own property.
         1. Let _len_ be the ExpectedArgumentCount of _ParameterList_.
-        1. Perform ! DefinePropertyOrThrow(_F_, `"length"`, PropertyDescriptor{[[Value]]: _len_, [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true*}).
+        1. Perform SetFunctionLength(_F_, _len_).
```

Comment:
I disagree with the above; this should be prefixed with !.

Noise is less of a concern to me than not knowing whether an abstract operation might throw or not.

Reply:
Would you prefix *all* currently-unprefixed operation-invocations with "!"? There's about 2300 of them.

Alternatively, we could come up with a naming convention that distinguished those operations that can return abrupt completions from those that can't. That would convey the information you want to see at the invocation-point, in a maybe less obtrusive way.

Personally, I think I'd prefer if information about what the operation can return were only a click or mouse-hover away.

Reply:
(I wonder if we should move this discussion to a separate issue.)

Reply:
Yes, i would, and yes, I’d say that’d be a separate issue.

Reply:
That there’s 2300 mean it’d be a tedious conversion; when I’m reading a specific part of the spec there wouldn’t be 2300 - there’d be just enough so I’d have to jump elsewhere as rarely as possible.

Reply:
Separate issue now created: #1097